lib/editor-linter.coffee

```diff
@@ -1,102 +1,40 @@
-{CompositeDisposable, Emitter, Range} = require 'atom'
-Helpers = require './helpers'
+{TextEditor, Emitter, CompositeDisposable} = require('atom')
 
 class EditorLinter
-  constructor: (@linter, @editor) ->
-    @status = true
-    @messages = new Map
-    @inProgress = false
-    @inProgressFly = false
-
-    if @editor is atom.workspace.getActiveTextEditor()
-      @linter.views.updateLineMessages(true)
-
+  constructor: (@editor) ->
+    throw new Error("Given editor isn't really an editor") unless @editor instanceof TextEditor
     @emitter = new Emitter
     @subscriptions = new CompositeDisposable
+    @subscriptions.add @editor.onDidDestroy =>
+      @emitter.emit 'did-destroy'
 
-    @subscriptions.add(
-      @editor.onDidSave => @lint(false)
-    )
-    @subscriptions.add(
-      @editor.onDidChangeCursorPosition ({oldBufferPosition, newBufferPosition}) =>
-        if newBufferPosition.row isnt oldBufferPosition.row
-          @linter.views.updateLineMessages(true)
-        @linter.views.renderBubble(newBufferPosition)
-    )
-    @subscriptions.add(
-      @editor.onDidStopChanging => @lint(true) if @linter.lintOnFly
-    )
+    @subscriptions.add @editor.onDidSave => @emitter.emit('should-lint', false)
+    @subscriptions.add @editor.onDidChangeCursorPosition ({oldBufferPosition, newBufferPosition}) =>
+      if newBufferPosition.row isnt oldBufferPosition.row
+        @emitter.emit('should-update-line-messages')
+      @emitter.emit('should-update-bubble')
+    # The onDidStopChanging callbacks are invoked immediately on creation, We are just
+    # gonna wait until a bit to get real events
+    setImmediate =>
+      @subscriptions.add @editor.onDidStopChanging => setImmediate => @emitter.emit('should-lint', true)
```

Comment:
yup, which is why I didn't push the button.

I'm sick tonight so I won't be reviewing the others. Maybe someone else in @atom-community/linter-contributors can review #737 and #738?